Dockerfile

```diff
@@ -25,15 +25,39 @@ ENV PATH=${PATH}:/conda/bin
 SHELL ["/bin/bash", "-c"]
 
 # Build cuDF conda env
-ADD conda /cudf/conda
-RUN conda env create --name cudf --file /cudf/conda/environments/dev_py35.yml
+ARG PYTHON_VERSION
+ENV PYTHON_VERSION=$PYTHON_VERSION
+ARG NUMBA_VERSION
+ENV NUMBA_VERSION=$NUMBA_VERSION
+ARG NUMPY_VERSION
+ENV NUMPY_VERSION=$NUMPY_VERSION
+ARG PANDAS_VERSION
+ENV PANDAS_VERSION=$PANDAS_VERSION
+ARG PYARROW_VERSION
+ENV PYARROW_VERSION=$PYARROW_VERSION
+ARG CYTHON_VERSION
+ENV CYTHON_VERSION=$CYTHON_VERSION
+ARG CMAKE_VERSION
+ENV CMAKE_VERSION=$CMAKE_VERSION
+ARG CUDF_REPO=https://github.com/rapidsai/cudf
+ENV CUDF_REPO=$CUDF_REPO
+ARG CUDF_BRANCH=master
+ENV CUDF_BRANCH=$CUDF_BRANCH
 
-# Preserved for users who currently use these build-args
-# Clone cuDF repo
-#ARG CUDF_REPO=https://github.com/rapidsai/cudf
-#ARG CUDF_BRANCH=master
-#RUN git clone --recurse-submodules -b ${CUDF_BRANCH} ${CUDF_REPO} /cudf
-ADD cpp /cudf/cpp
+# Add everything from the local build context
+ADD . /cudf/
```

Comment:
This approach results in "simple" changes to cudf/python (code OR docstring formatting) causing entire image rebuilds, a very long process.

Yes, it's "ugly" to adding folders one at a time in the Dockerfile, but it lets you re-build small image changes iteratively (<1m to rebuild cudf/python bindings and/or docs, compared to several minutes for an entire image).

Edit: Testing now to confirm that a doc change rebuilds the whole container

Reply:
Merging this as is, raise an issue regarding investigating improving the context transfer time / build time.